src/python/tritonfrontend/CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+include(FetchContent)
+
+FetchContent_Declare(
+    pybind11
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
+    GIT_TAG v2.13.1
+    GIT_SHALLOW ON
+)
+
+FetchContent_MakeAvailable(pybind11)
+# ================================= END =====================================
+
+# ================== Collect the Dependencies ===============================
+set(
+  PYTHON_FRONTEND_BINDING_DEPS
+  ../../shared_memory_manager.h
+  ../../shared_memory_manager.cc
+  ../../data_compressor.h
+  ../../common.h
+  ../../common.cc
+  ../../restricted_features.h
+  ../../tracer.h
+  $<$<BOOL:${TRITON_ENABLE_TRACING}>:../../tracer.cc>
+  ../../classification.cc
+)
+
+set(PY_BINDING_DEPENDENCY_LIBS
+      triton-common-json
+      triton-common-logging
+      triton-core-serverapi
+      triton-core-serverstub
+  )
+
+# Conditional Linking Based on Flags
+if(${TRITON_ENABLE_HTTP})
+  list(APPEND PY_BINDING_DEPENDENCY_LIBS
+      http-endpoint-library
+    )
+endif()
+
+if(${TRITON_ENABLE_GRPC})
+  list(APPEND PY_BINDING_DEPENDENCY_LIBS
+      grpc-endpoint-library
+  )
+endif()
+
+if(${TRITON_ENABLE_GPU})
+  find_package(CUDAToolkit REQUIRED)
+  list(APPEND PY_BINDING_DEPENDENCY_LIBS
+      CUDA::cudart
+  )
+endif()
+
+if(${TRITON_ENABLE_TRACING} OR ${TRITON_ENABLE_STATS})
+  message("TRACING/STATS IS CURRENTLY NOT SUPPORTED.")
+  find_package(absl CONFIG REQUIRED)
+  find_package(CURL CONFIG REQUIRED)
+  find_package(nlohmann_json CONFIG REQUIRED)
+  find_package(opentelemetry-cpp CONFIG REQUIRED)
+  list(APPEND PY_BINDING_DEPENDENCY_LIBS
+    tracing-library
+  )
+endif()
+
+# ===================== End of Collection ===================================
+
+
+# ================== Create Python Frontend Bindings ========================
+set(
+  PYTHON_FRONTEND_BINDING_SRCS
+  _c/tritonfrontend.h
+  _c/tritonfrontend_pybind.cc
+)
+
+pybind11_add_module(
+  py-bindings
+  MODULE
+  ${PYTHON_FRONTEND_BINDING_DEPS}
+  ${PYTHON_FRONTEND_BINDING_SRCS}
+)
+
+target_include_directories(py-bindings PRIVATE ${CMAKE_SOURCE_DIR}/src)
+
+target_link_libraries(
+    py-bindings
+    PRIVATE
+      ${PY_BINDING_DEPENDENCY_LIBS}
+)
+
+if(${TRITON_ENABLE_HTTP})
+  target_compile_definitions(
+    py-bindings
+    PRIVATE TRITON_ENABLE_HTTP=1
+  )
+endif()
+
+if(${TRITON_ENABLE_GRPC})
+  target_compile_definitions(
+    py-bindings
+    PRIVATE TRITON_ENABLE_GRPC=1
+  )
+endif()
+
+if(${TRITON_ENABLE_GPU})
+  target_compile_definitions(
+    py-bindings
+    PRIVATE TRITON_ENABLE_GPU=1
+    PRIVATE TRITON_MIN_COMPUTE_CAPABILITY=${TRITON_MIN_COMPUTE_CAPABILITY}
+  )
+endif()
+
+if(${TRITON_ENABLE_TRACING})
+  target_include_directories(
+      py-bindings
+      PRIVATE ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
+    )
+  target_compile_definitions(
+    py-bindings
+    PRIVATE TRITON_ENABLE_TRACING=1
+  )
+endif()
+
+if(${TRITON_ENABLE_STATS})
+  target_include_directories(
+      py-bindings
+      PRIVATE ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
```

Comment:
why stats involves opentelemetry?

Reply:
Initially added because CMake has a conditional check for `{TRITON_ENABLE_TRACING} AND NOT {TRITON_ENABLE_STATS}`. But stats being enabled does not imply tracing is enabled. 
So, removed this block of code. Thank you for pointing it out.